index.js

```diff
@@ -37,12 +37,14 @@ module.exports = function headless(options, startnum, callback) {
   }
 
   findFreeServernum(startnum, function(servernum) {
-    if (!options) {
-      var childProcess = spawn('Xvfb', [':' + servernum]);
-    }
-    else {
-      var childProcess = spawn('Xvfb', [':' + servernum , '-screen' , '0' , options.display.width + 'x' + options.display.height + 'x' + (options.display.depth || 16) ]);
+    options = options || {}
+
+    var args = options.args || []
+    var display = []
+    if (options.display) {
+      display = ['-screen', '0', options.display.width + 'x' + options.display.height + 'x' + (options.display.depth || 16)]
     }
+    var childProcess = spawn('Xvfb', [':' + servernum].concat(display).concat(args));
```

Comment:
For readability I'd create an `args` variable, also `display` can be written like `var display = options.display ? ['-screen', '0', options.display.width + 'x' + options.display.height + 'x' + (options.display.depth || 16)] : []`.

So something like (I haven't actually tested this so sorry if this doesn't run)

```js
var display = options.display ? ['-screen', '0', options.display.width + 'x' + options.display.height + 'x' + (options.display.depth || 16)] : [];
var args = [':' + servernum].concat(display).concat(options.args || []);
```